contracts/bonding/BondingManager.sol

```diff
@@ -1441,6 +1484,8 @@ contract BondingManager is ManagerProxyTarget, IBondingManager {
         // Rewards are bonded by default
         del.bondedAmount = currentBondedAmount;
         del.fees = currentFees;
+
+        checkpointBondingState(_delegator, del, transcoders[_delegator]);
```

Comment:
This will result in 2 calls to `checkpointBondingState()` in every bonding related function that uses the `autoClaimEarnings()` modifier or the `_autoClaimEarnings()` internal function. Could we add this line to just the end of `claimEarnings()` instead to avoid this?